### ISSUE TITLE
token-2022: `get_account_data_size` accepts mint extensions

### ISSUE THREAD

**Issue by joncinque** [solana-labs/solana-program-library#4587]
#### Problem

The `get_account_data_size` instruction accepts all any extension type, including `Uninitialized` and mint extensions. It will happily calculate the number, even if it isn't possible to initialize an account with those extensions later.

#### Solution

For safety, return an error for `Uninitialized` or any extension types that describe mints.

cc @ryoqun

**Comment by joncinque** [comment]
Certainly! You'll just need to make a change to check the extension types passed in https://github.com/solana-labs/solana-program-library/blob/727fdd7ecf22d9e3573e4014308d310b5d08a149/token/program-2022/src/processor.rs#L1229-L1245

This is is very similar to https://github.com/solana-labs/solana-program-library/blob/727fdd7ecf22d9e3573e4014308d310b5d08a149/token/program-2022/src/extension/reallocate.rs#L50

And you'll need more test cases in https://github.com/solana-labs/solana-program-library/blob/727fdd7ecf22d9e3573e4014308d310b5d08a149/token/program-2022/src/processor.rs#L7351

**Comment by dhruvdabhi101** [comment]
If no one is currently working on this issue, then I might give a try.
